R/formatter.r

```diff
@@ -182,25 +173,50 @@ dollar <- function(x, prefix = "$", suffix = "",
   }
 }
 
-#' Percent formatter: multiply by one hundred and display percent sign.
-#'
-#' @return a function with single parameter x, a numeric vector, that
-#'   returns a character vector
-#' @param x a numeric vector to format
+
+#' `percent_format` and `percent` multiply values by one hundred and
+#' display percent sign.
 #' @export
+#' @rdname number_format
 #' @examples
+#' # percent() and percent_format()
 #' percent_format()(runif(10))
 #' percent(runif(10))
-#' percent(runif(10, 1, 10))
-percent_format <- function() {
-  function(x) percent(x)
+#'
+#' per_mille <- percent_format(
+#'   scale = 1000,
+#'   suffix = "\u2030",
+#'   accuracy = .1
+#' )
+#' per_mille(.1234)
+#'
+#' french_percent <- percent_format(
+#'   decimal.mark = ",",
+#'   suffix = " %"
+#' )
+#' french_percent(runif(10))
+#'
+percent_format <- function(accuracy = NULL, scale = 100, suffix = "%", ...) {
```

Comment:
Is there a reason to make scale or suffix parameters?

Reply:
Oh, I see for `per_mille`. I think it would be simpler to continue to use `number_format()` for that example.